dm/syncer/schema.go

```diff
@@ -50,29 +51,53 @@ func (s *Syncer) OperateSchema(ctx context.Context, req *pb.OperateWorkerSchemaR
 		for i, schema := range allSchema {
 			schemaList[i] = schema.Name.String()
 		}
-		schemaListJSON, err := json.Marshal(schemaList)
-		if err != nil {
-			return "", terror.ErrSchemaTrackerMarshalJSON.Delegate(err, schemaList)
+		schemaListJSON, err2 := json.Marshal(schemaList)
+		if err2 != nil {
+			return "", terror.ErrSchemaTrackerMarshalJSON.Delegate(err2, schemaList)
 		}
-		return string(schemaListJSON), err
+		return string(schemaListJSON), err2
 	case pb.SchemaOp_ListTable:
-		tables, err := s.schemaTracker.ListSchemaTables(req.Database)
-		if err != nil {
-			return "", err
+		tables, err2 := s.schemaTracker.ListSchemaTables(req.Database)
+		if err2 != nil {
+			return "", err2
 		}
-		tableListJSON, err := json.Marshal(tables)
-		if err != nil {
-			return "", terror.ErrSchemaTrackerMarshalJSON.Delegate(err, tables)
+		tableListJSON, err2 := json.Marshal(tables)
+		if err2 != nil {
+			return "", terror.ErrSchemaTrackerMarshalJSON.Delegate(err2, tables)
 		}
-		return string(tableListJSON), err
+		return string(tableListJSON), err2
 	case pb.SchemaOp_GetSchema:
 		// when task is paused, schemaTracker is closed. We get the table structure from checkpoint.
 		ti := s.checkpoint.GetTableInfo(req.Database, req.Table)
 		if ti == nil {
 			s.tctx.L().Info("table schema is not in checkpoint, fetch from downstream",
 				zap.String("table", sourceTable.String()))
 			targetTable := s.route(sourceTable)
-			return dbconn.GetTableCreateSQL(s.tctx.WithContext(ctx), s.downstreamTrackConn, targetTable.String())
```

Comment:
I think we don't need to remove the dbconn.GetTableCreateSQL() calling and replace it with `SHOW CREATE TABLE` calling.  But rather: 
```golang
			result, err2 := dbconn.GetTableCreateSQL(s.tctx.WithContext(ctx), s.downstreamTrackConn, targetTable.String())
			result = strings.Replace(result, fmt.Sprintf("CREATE TABLE %s", quotes.QuoteName(targetTable.Name)), fmt.Sprintf("CREATE TABLE %s", quotes.QuoteName(sourceTable.Name)), 1)
			return utils.CreateTableSQLToOneRow(result), err2
```

Reply:
dbconn.GetTableCreateSQL() is not included in the old-version branch. Moreover, the committed logic is not equal to dbconn.GetTableCreateSQL() since conn.QuerySQL() between the versions are of different APIs.

Reply:
OK.  Can we reuse `(*schema.Tracker).GetCreateTable()` ?  In my opinion, this PR mainly changes the table name from target table name to source table name, and changes the sql into a single line ( not in this version ).  Maybe : 
```golang
targetTable := s.route(sourceTable)
result, err2 := s.schemaTracker.GetCreateTable(ctx, targetTable)
result = strings.Replace(result, fmt.Sprintf("CREATE TABLE %s", quotes.QuoteName(targetTable.Name)), fmt.Sprintf("CREATE TABLE %s", quotes.QuoteName(sourceTable.Name)), 1)
return result, err2
```

Reply:
Actually I am not sure about this. In my opinion, getting the table from schema tracker is an incorrect version. Getting schema can only be operated when task is paused. However, when the task is paused, the schema tracker is down and nothing would return. I have tried to build the original version and observed this phenomenon.